src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.Helper.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+using System;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    internal abstract partial class AbstractBuildEventValueProvider
+    {
+        public abstract class Helper
+        {
+            private const string _execTaskName = "Exec";
+            private const string _commandString = "Command";
+
+            protected Helper(string buildEventString,
+                   string targetNameString,
+                   Func<ProjectTargetElement, string> getTargetString,
+                   Action<ProjectTargetElement> setTargetDependencies)
+            {
+                BuildEventString = buildEventString;
+                TargetNameString = targetNameString;
+                GetTargetString = getTargetString;
+                SetTargetDependencies = setTargetDependencies;
+            }
+
+            private Func<ProjectTargetElement, string> GetTargetString { get; }
+            private Action<ProjectTargetElement> SetTargetDependencies { get; }
+            private string BuildEventString { get; }
+            private string TargetNameString { get; }
+
+            public string GetProperty(ProjectRootElement projectXml)
+            {
+                var result = FindExecTaskInTargets(projectXml);
+
+                if (result.success == false)
+                {
+                    return null;
+                }
+
+                if (result.execTask.Parameters.TryGetValue(_commandString, out var commandText))
+                {
+                    return commandText;
+                }
+
+                return null; //unreachable
```

Comment:
Better to add a Assert fail so that we get notified when this happens rather than silently passing